tools/kraken.py

```diff
@@ -55,51 +55,118 @@ def build(self, db, options=None, option_string=None):
         self.execute('kraken-build', db, db, options=options,
                      option_string=option_string)
 
-    def pipeline(self, inBam, db, outReport=None, outReads=None, filterThreshold=None, numThreads=None):
-
-        with util.file.fifo(2) as (fastq1_pipe, fastq2_pipe):
-            # do not convert this to samtools bam2fq unless we can figure out how to replicate
-            # the clipping functionality of Picard SamToFastq
-            picard = tools.picard.SamToFastqTool()
-            picard_opts = {
-                'CLIPPING_ATTRIBUTE': tools.picard.SamToFastqTool.illumina_clipping_attribute,
-                'CLIPPING_ACTION': 'X'
-            }
-            picard.execute(inBam, fastq1_pipe, fastq2_pipe,
-                           picardOptions=tools.picard.PicardTools.dict_to_picard_opts(picard_opts),
-                           JVMmemory=picard.jvmMemDefault,
-                           background=True)
-
-            opts = {
-                '--threads': util.misc.sanitize_thread_count(numThreads),
-            }
+
+    def db_opts(self, db, threads):
+        '''Determine kraken command line options based on db string'''
+
+        env = os.environ.copy()
+        def s3_psub(path):
+            cmd = 'aws s3 cp {} -'.format(path)
+            if path.endswith('.bz2'):
+                cmd += ' | lbzip2 -n {threads} -d'.format(threads=threads)
+            elif path.endswith('.gz'):
+                cmd += ' | pigz -p {threads} -d'.format(threads=threads)
```

Comment:
pigz doesn't exist in the conda package, though it does exist in the docker image.. we can make an effort later to harmonize some of this code later on

Reply:
s/lbzip2/bzip2/

Reply:
`lbzip2` is the parallelized version of `bzip2`

Reply:
ok I'll add it to requirements-conda.txt in my pr in a minute

Reply:
There's also `pbzip2`, but I've seen some benchmarks that show `lbzip2` is a bit faster. My benchmark are still running.